### PR TITLE
Merge pull request #4 from wejago/H2I-3_add_grafana

### PR DESCRIPTION
**Pull request by emiliyank**
Setup docker compose

### CODE REVIEW

File: deployment/.env.example [hunk 1]
```diff
@@ -3,3 +3,6 @@ MQTT_PASSWORD=sample_password
 DOCKER_INFLUXDB_INIT_USERNAME=sample_user
 DOCKER_INFLUXDB_INIT_PASSWORD=sample_password
 DOCKER_INFLUXDB_INIT_ADMIN_TOKEN=sample_token
+GF_SECURITY_ADMIN_USER=sample_user
```
Comment (burberius) [inline deployment/.env.example:6]: Where are these 3 variables used?
Please also add them the the containers, so they will be shown as missing if .env is missing.

File: deployment/docker-compose.yml [hunk 2]
```diff
@@ -38,6 +38,8 @@ services:
   influxdb:
     container_name: influxdb
     image: influxdb:latest
+    volumes:
+      - "./influxdb:/var/lib/influxdb2"
```
Comment (burberius) [inline deployment/docker-compose.yml:42]: Add to the readme, that the directory needs to be created for the DB data.

**Review by burberius — CHANGES_REQUESTED** [review]

File: deployment/.env.example (hunk 1 above)
Comment (emiliyank) [reply deployment/.env.example:6]: https://grafana.com/docs/grafana/latest/setup-grafana/configure-grafana/#override-configuration-with-environment-variables - they are not used in other places of the code but we need them to set different user & pass

Comment (emiliyank) [reply deployment/.env.example:6]: INFLUX_REMOTE_URL is used for the datasource setup

File: deployment/docker-compose.yml (hunk 2 above)
Comment (emiliyank) [reply deployment/docker-compose.yml:42]: Sorry I didn't get this one. Can you please clarify

Comment (burberius) [reply deployment/docker-compose.yml:42]: Put in the README.md that for the deployment you need to create that influxdb folder.

File: deployment/.env.example (hunk 1 above)
Comment (cupakob) [inline deployment/.env.example:6]: `GF` should means Grafana? If yes - then change to `GRAFANA` please

File: deployment/docker-compose.yml [hunk 3]
```diff
@@ -53,5 +55,28 @@ services:
...
+      - influxdb
+    ports:
+      - 3000:3000
+    volumes:
+      - "./grafana/provisioning/dashboards:/etc/grafana/provisioning/dashboards"
+      - "./grafana/provisioning/datasources:/etc/grafana/provisioning/datasources"
+      - "./grafana/dashboards:/var/lib/grafana/dashboards"
+    environment:
+      - DOCKER_INFLUXDB_INIT_ORG=wejago
+      - DOCKER_INFLUXDB_INIT_BUCKET=bucket1
```
Comment (cupakob) [inline deployment/docker-compose.yml]: should we rename the bucket? something like "hichi" or similar, beacause we are collecting data from a hichi device

File: deployment/grafana/dashboards/sample-dashboard.json [hunk 4]
```diff
@@ -0,0 +1,638 @@
...
+                {
+                    "datasource": {
+                        "type": "influxdb",
+                        "uid": "P951FEA4DE68E13C5"
+                    },
+                    "query": "import \"date\"\r\nimport \"timezone\"\r\noption location = timezone.location(name: \"Europe/Berlin\")\r\n\r\nfrom(bucket: \"bucket1\")\r\n  |> range(start: -1y)\r\n  |> filter(fn: (r) =>\r\n    r._measurement == \"sensor\"  and\r\n    r._field == \"currentConsumption(16_7_0)\"\r\n  )\r\n  |> map(fn: (r) => ({ r with hour: date.hour(t: r._time) }))\r\n  |> group(columns: [\"hour\"], mode:\"by\")\r\n  |> mean(column: \"_value\") \r\n  |> group()\r\n  |> sort(columns: [\"hour\"])",
+                    "refId": "A"
+                }
+            ],
+            "title": "AVG Hourly Consumption - past 12 months",
```
Comment (cupakob) [inline deployment/grafana/dashboards/sample-dashboard.json]: change `AVG` to `Average` to stay consistent.

File: deployment/.env.example (hunk 1 above)
Comment (emiliyank) [reply deployment/.env.example:6]: These names are defined by Grafana so we cannot change them - https://grafana.com/docs/grafana/latest/setup-grafana/configure-grafana/#override-configuration-with-environment-variables

File: deployment/docker-compose.yml (hunk 2 above)
Comment (emiliyank) [reply deployment/docker-compose.yml:42]: It is created by the `docker-compose` command. I think you don't have to create it manually

**Review by burberius — APPROVED** [review]

File: deployment/docker-compose.yml (hunk 3 above)
Comment (burberius) [reply deployment/docker-compose.yml]: I would leave it general, my parents device is not called hichi. Hichi is only the nickname of the guy that assembled and sold your device on ebay!

**Review by cupakob — APPROVED** [review]

**Pull request merged**